typing/types.mli

```diff
@@ -331,6 +333,12 @@ and constructor_arguments =
   | Cstr_tuple of type_expr list
   | Cstr_record of label_declaration list
 
+and unboxed_status =
```

Comment:
Shouldn't this be a record type such as `{boxed: bool; explicit: bool}`?  This would remove the need for helper functions in Btype.